tests/unit/models/TerraformData.spec.js

```diff
@@ -0,0 +1,194 @@
+        .toEqual('test_value');
+      expect(terraformData.getVariableValue('local.test_variable2'))
+        .toBeNull();
+    });
+
+    it('Should return null on bad variable reference', () => {
+      const variable = new TerraformVariable({
+        name: 'test_variable',
+        value: 'test_value',
+        category: 'variable',
+      });
+
+      terraformData.variables.push(variable);
+      expect(terraformData.getVariableValue('var.test_variable.error'))
+        .toBeNull();
+    });
+  });
+
+  describe('Test method: getAttributeValue', () => {
+    it('Should return the value of a variable', () => {
+      const variable = new TerraformVariable({
+        name: 'test_variable',
+        value: 'test_value',
+        category: 'variable',
+      });
+      const attribute = new TerraformComponentAttribute({
+        value: 'var.test_variable',
+      });
+
+      terraformData.variables.push(variable);
+      expect(terraformData.getAttributeValue(attribute))
+        .toEqual('test_value');
+    });
+
+    it('Should return the ID of components links', () => {
+      const definition = new TerraformComponentAttributeDefinition({
+        type: 'Link',
+      });
+      const attribute = new TerraformComponentAttribute({
+        definition,
+        value: [
+          'test_resource.test_id_1.test_value',
+          'test_resource.test_id_2.test_value',
+        ],
+      });
+
+      expect(terraformData.getAttributeValue(attribute))
+        .toEqual(['test_id_1', 'test_id_2']);
+    });
+
+    it('Should return the ID of a component reference', () => {
+      const definition = new TerraformComponentAttributeDefinition({
+        type: 'Reference',
+      });
+      const attribute = new TerraformComponentAttribute({
+        definition,
+        value: 'test_resource.test_id.test_value',
+      });
+
+      expect(terraformData.getAttributeValue(attribute))
+        .toEqual('test_id');
+    });
+
+    it('Should return the attribute value', () => {
+      const attribute = new TerraformComponentAttribute({
+        value: 'test_value',
+      });
+
+      expect(terraformData.getAttributeValue(attribute))
+        .toEqual('test_value');
+    });
+  });
+
+  describe('Test method: getComponentIdFromValue', () => {
+    it('Should return the id of a resource reference', () => {
+      expect(terraformData.getComponentIdFromValue('test_resource.test_id.test_value'))
+        .toEqual('test_id');
+    });
+
+    it('Should return the id of a data reference', () => {
+      expect(terraformData.getComponentIdFromValue('data.test_resource.test_id.test_value'))
+        .toEqual('test_id');
+    });
+
+    it('Should return the whole value', () => {
+      expect(terraformData.getComponentIdFromValue('test_value'))
+        .toEqual('test_value');
+    });
+  });
+
+  describe('Test method: getLinkedComponentsIds', () => {
+    it('Should return the ID of a variable', () => {
```

Comment:
`it('Should return an array containing the ID of a variable', () => {`

Reply:
done